lld/ELF/Arch/X86_64.cpp

```diff
@@ -486,18 +490,26 @@ void X86_64::relaxTlsGdToLe(uint8_t *loc, const Relocation &rel,
     // The original code used a pc relative relocation and so we have to
     // compensate for the -4 in had in the addend.
     write32le(loc + 8, val + 4);
-  } else if (rel.type == R_X86_64_GOTPC32_TLSDESC) {
+  } else if (rel.type == R_X86_64_GOTPC32_TLSDESC ||
+             rel.type == R_X86_64_CODE_4_GOTPC32_TLSDESC) {
     // Convert leaq x@tlsdesc(%rip), %REG to movq $x@tpoff, %REG.
     if ((loc[-3] & 0xfb) != 0x48 || loc[-2] != 0x8d ||
```

Comment:
It seems you're missing some tests.  The `(loc[-3] & 0xfb) != 0x48` is used to check the X and B bit for REX prefix, it would fail for REX2 prefix.

Reply:
Never mind. I got something wrong. It works for REX2 too if we don't support
```
{rex2} leaq x@tlsdesc(%rip), %r8 
```
for this relocation.

Currently, compiler won't emit such code.